src/main/java/org/cbioportal/web/columnar/BasicDataBinner.java

```diff
@@ -84,11 +98,22 @@ public <T extends DataBinCountFilter, S extends DataBinFilter, U extends DataBin
         Map<String, ClinicalDataType> attributeDatatypeMap;
         switch (dataBinCountFilter) {
             // TODO: first case is to support clinical data, but clinical data is not using this now. We should update controller to use this method later
-            case ClinicalDataBinCountFilter clinicalDataBinCountFilter -> {
+            case ClinicalDataBinCountFilter clinicalDataBinCountFilter when customDataService.getCustomDataSessions(uniqueKeys).isEmpty() -> {
                 unfilteredClinicalDataCounts = studyViewColumnarService.getClinicalDataCounts(partialFilter, uniqueKeys);
                 filteredClinicalDataCounts = studyViewColumnarService.getClinicalDataCounts(studyViewFilter, uniqueKeys);
                 attributeDatatypeMap = studyViewColumnarService.getClinicalAttributeDatatypeMap();
             }
+            case ClinicalDataBinCountFilter clinicalDataBinCountFilter -> {
```

Comment:
You might need to add the other case when its not empty or it might fall through? or did you test this?

Reply:
yea i've tested this and when its not empty and it looks fine to me